### ISSUE TITLE
[BUG] TMC2209 not working anymore after version upgrade

### ISSUE THREAD

**Issue by igna09** [MarlinFirmware/Marlin#26722]
### Did you test the latest `bugfix-2.1.x` code?

Yes, and the problem still exists.

### Bug Description

I used my printer with Marlin 2.0.5.3 and TMC2209 (X, Y, Z, E) without any issues
Today i decided to upgrade to bugfix 2.1.x porting old configurations
I noticed that homing (sensorless) was not working anymore
I executed M122 I S0 and noticed two strange behaviours:
- despite S0 flag it continue to output values instead of stopping
- current values was wrong
I tryed with M502 to restore configurations from configuration.h and configuration_adv.h but nothing changed

So I decided to downgrade to 2.0.5.3 and everything is working again

There is some breaking change (about trinamic) upgrading from 2.0.5.3 that I'm missing or is this a bug?

### Expected behavior

it should load trinamic settings (#if HAS_TRINAMIC_CONFIG) from configuration files

### Actual behavior

it load default trinamic settings

### Version of Marlin Firmware

bugfix 2.1.x

### Bed Leveling

None

### Your Slicer

None

### Host Software

None

### Don't forget to include

- [X] A ZIP file containing your `Configuration.h` and `Configuration_adv.h`.

### Additional information & file uploads

[Configuration_2_0_5_3.zip](https://github.com/MarlinFirmware/Marlin/files/14027662/Configuration_2_0_5_3.zip)
[Configuration_bugfix2_1_x.zip](https://github.com/MarlinFirmware/Marlin/files/14027675/Configuration_bugfix2_1_x.zip)

**Comment by igna09** [comment]
I also checked that connection is working, in both versions i can read registers correctly

**Comment by sargonphin** [comment]
Could you be more precise about what's not working? Also, try a minimal configuration from a fresh configuration file (just enough to run homing) and see if homing works. Enable DEBUG_LEVELING_FEATURE, then M111 S247 and then run G28 to see if you can spot anything obviously wrong ^_^

**Comment by phazei** [comment]
Same for me.  It was working in 2.0.5.3 then I moved to 2.1 and it's not triggering, even if I set the sensitivity to max.  I went through old versions and it seems it doesn't work from 2.0.8
Not sure if what is supposed to be set for that has changed or not.

**Comment by Tannoo** [comment]
My build, `STRING_DISTRIBUTION_DATE "2024-01-24"` of `SHORT_BUILD_VERSION "bugfix-2.1.x"` using sensorless homing for both X and Y on TMC2209 drivers.

I have no issues.

```
Send: M122
Recv: 		X	Y	Z	E
Recv: Address		0	0	0	0
Recv: Enabled		false	false	false	false
Recv: Set current	800	1300	1100	800
Recv: RMS current	795	1270	1049	795
Recv: MAX current	1121	1791	1479	1121
Recv: Run current	25/31	22/31	18/31	25/31
Recv: Hold current	12/31	11/31	9/31	12/31
Recv: CS actual	12/31	11/31	9/31	12/31
Recv: PWM scale
Recv: vsense		1=.18	0=.325	0=.325	1=.18
Recv: stealthChop	true	true	true	true
Recv: msteps		64	64	64	16
Recv: interp		true	true	true	true
Recv: tstep		max	max	max	max
Recv: PWM thresh.
Recv: [mm/s]
Recv: OT prewarn	false	false	false	false
Recv: triggered
Recv:  OTP		false	false	false	false
Recv: pwm scale sum	16	17	15	16
Recv: pwm scale auto	2	4	4	2
Recv: pwm offset auto	36	36	36	36
Recv: pwm grad auto	14	14	14	14
Recv: off time	4	4	4	4
Recv: blank time	24	24	24	24
Recv: hysteresis
Recv:  -end		2	2	2	2
Recv:  -start		1	1	1	1
Recv: Stallguard thrs	66	80	0	0
Recv: uStep count	10	54	2	8
Recv: DRVSTATUS	X	Y	Z	E
Recv: sg_result	0	0	0	0
Recv: stst
Recv: olb
Recv: ola
Recv: s2gb
Recv: s2ga
Recv: otpw
Recv: ot
Recv: 157C
Recv: 150C
Recv: 143C
Recv: 120C
Recv: s2vsa
Recv: s2vsb
Recv: Driver registers:
Recv: 		X	0xC0:0C:00:00
Recv: 		Y	0xC0:0B:00:00
Recv: 		Z	0xC0:09:00:00
Recv: 		E	0xC0:0C:00:00
Recv: 
Recv: 
Recv: Testing X connection... OK
Recv: Testing Y connection... OK
Recv: Testing Z connection... OK
Recv: Testing E connection... OK
```

[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/14073340/Configuration.zip)
[Configuration_adv.zip](https://github.com/MarlinFirmware/Marlin/files/14073341/Configuration_adv.zip)

**Comment by phazei** [comment]
It took a lot of testing, but I did finally get mine working.  SKR1.4, TMC2209, BL Touch, CoreXY, Marlin 2.1.2.1:
[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/14074554/Configuration.zip)

I particularly had issues with stealthChop and hybrid_threshold.  Depending on what they were set at, the sensorless homing didn't work at all.  I had to disable them entirely, or lower my microsteps from 256 to 64.  Hopefully these files might help you.
